tests/mock_network/utils.rs

```diff
@@ -374,11 +374,16 @@ pub fn create_connected_nodes(env: &Environment, size: usize) -> Vec<TestNode> {
     for _ in 1..size {
         let config = TransportConfig::node().with_hard_coded_contact(endpoint);
         nodes.push(TestNode::builder(env).transport_config(config).create());
-
-        poll_until(env, &mut nodes, |nodes| node_joined(nodes, nodes.len() - 1));
-        verify_invariants_for_nodes(&env, &nodes);
     }
 
+    poll_until(env, &mut nodes, |nodes| {
+        all_nodes_joined(nodes, 0..nodes.len())
+    });
+    poll_until(env, &mut nodes, |nodes| {
+        elder_count_reached(nodes, &Prefix::default(), env.elder_size().min(size))
```

Comment:
why not just poll this second? seems the above poll is duplicated?

Reply:
Because we need both: all the new nodes joining and the section reaching the minimal size of elders. Neither one implies the other.

Reply:
ah, I see what they means.